daemon/delete.go

```diff
@@ -115,13 +115,9 @@ func (daemon *Daemon) rm(container *Container, forceRemove bool) (err error) {
 		}
 	}()
 
-	if _, err := daemon.containerGraphDB.Purge(container.ID); err != nil {
-		logrus.Debugf("Unable to remove container from link graph: %s", err)
-	}
-
 	metadata, err := daemon.layerStore.DeleteMount(container.ID)
 	layer.LogReleaseMetadata(metadata)
-	if err != nil {
+	if err != nil && err != layer.ErrMountDoesNotExist {
```

Comment:
This change fixed the failed test described at https://github.com/docker/docker/pull/18308#issuecomment-160657007  
https://jenkins.dockerproject.org/job/Docker-PRs/20223/console